src/tribol/mesh/CouplingScheme.cpp

```diff
@@ -1476,64 +1529,66 @@ void CouplingScheme::computeCommonPlaneTimeStep(real &dt)
-         dt1 = (dt1_check1) ? alpha * max_delta1 / v1_dot_n1 : dt1;
-         dt2 = (dt2_check1) ? alpha * max_delta2 / v2_dot_n2 : dt2;
-
-         // Keep debug print statements. This routine is still in the testing phase
-         //std::cout << "dt1_check1, delta1 and v1_dot_n1: " << dt1_check1 << ", " << max_delta1 << ", " << v1_dot_n1 << std::endl;
-         //std::cout << "dt2_check1, delta2 and v2_dot_n2: " << dt2_check1 << ", " << max_delta2 << ", " << v2_dot_n2 << std::endl;
-         //std::cout << "dt1 and dt2: " << dt1 << ", " << dt2 << std::endl;
-
-         // update dt_temp1 only for positive dt1 and/or dt2
-         if (dt1 > 0.)
-         {
-            dt_temp1 = axom::utilities::min(dt_temp1, 
-                       axom::utilities::min(dt1, 1.e6));
-         }
-         if (dt2 > 0.)
-         {
-            dt_temp1 = axom::utilities::min(dt_temp1, 
-                       axom::utilities::min(1.e6, dt2));
-         }
-
-
-         if (dt1 < 0. || dt2 < 0.)
-         {
-            neg_dt_gap_msg = true;
-         }
+      if (plane.m_inContact)
+      {
+
+        // compute the difference between the 'face-gaps' and the max allowable 
+        // interpen as a function of element thickness. Note, we have to use the 
+        // gap projected onto the outward unit face-normal to check against the
+        // max allowable gap as a factor of the thickness in the element normal
+        // direction
+        RealT delta1 = max_delta1 - gap_f1_n1; // >0 not exceeding max allowable
+        RealT delta2 = max_delta2 + gap_f2_n2; // >0 not exceeding max allowable
+
+        auto exceed_max_gap1 = (delta1 < 0.) ? true : false;
+        auto exceed_max_gap2 = (delta2 < 0.) ? true : false;
+
+        // if velocity projection indicates further interpenetration, and the gaps
+        // EXCEED max allowable, then compute time step estimates to reduce overlap
+        dt1_check1 = (dt1_vel_check) ? exceed_max_gap1 : false;
+        dt2_check1 = (dt2_vel_check) ? exceed_max_gap2 : false;
+
+        msg[0] = exceed_max_gap1;
+        msg[1] = exceed_max_gap2;
+
+        // compute dt for face 1 and 2 based on the velocity and gap projections onto 
+        // the face-normals for faces where currect gap exceeds max allowable gap.
+        //
+        // NOTE:
+        //
+        // This calculation RESETS the current gap to be g = 0, and computes a timestep
+        // such that the velocity projection of the overlap-to-face projected overlap 
+        // centroid does not exceed the max allowable gap. 
+        //
+        // This avoid a timestep crash in the case that the current gap barely exceeds 
+        // the max allowable and also allows a soft contact response with interpen
+        // in excess of the max allowable gap without causing timestep crashes.
+        //
+        // v1_dot_n1 > 0 and v2_dot_n2 > 0 for further interpen
+        dt1 = (dt1_check1) ? -alpha * max_delta1 / v1_dot_n1 : dt1;
```

Comment:
The original calculation didn't have the `-` in front of the alpha. Please verify.

Reply:
Good catch, not sure where the `-` came from?  Probably missed from one of the merges with develop.

Reply:
Is there a way we can add test coverage to this?  Surprised the tests still passed.

Reply:
Create an issue to investigate why a test failure wasn't triggered